### PR TITLE
Update google-api-core dependency

### PR DESCRIPTION
**Pull request by Iinh**
Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title)
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
 #
-# This file is autogenerated by pip-compile with python 3.8
+# This file is autogenerated by pip-compile with python 3.9
```
Comment (Iinh) [inline requirements.txt:2]: I followed the steps laid out in the docs [here](https://mozilla.github.io/bigquery-etl/cookbooks/common_workflows/#adding-new-python-requirements) to manually update this dependency, but the content of the generated change in this file worries me a little bit... @wlach does this look ok to you?

Comment (wlach) [reply requirements.txt:2]: Yeah that's normal given differences in the python environment. I usually add these types of changes with `git add -p` to *just* add the changes that upgrade the dependency.

If you want to fix this without filing a whole new PR I would run something like `git revert HEAD^` in the branch, make the changes again (or copy them manually from the PR), and then make a new commit.

Comment (Iinh) [reply requirements.txt]: Thanks Will for the detailed instruction, as always!

**Review by wlach — APPROVED** [review]
Thanks for working through this one!

**Pull request merged**